src/main/kotlin/org/cafejojo/novelty/NoveltyAnalyzer.kt

```diff
@@ -0,0 +1,61 @@
+package org.cafejojo.novelty
+
+import mu.KotlinLogging
+import org.cafejojo.schaapi.models.project.JavaMavenProject
+import org.cafejojo.schaapi.models.project.MavenProject
+import java.io.File
+
+/**
+ * Determines the novelty of a test suite compared to another test suite.
+ *
+ * Expects a standard Maven project with the following folder setup:
+ *
+ * src
+ * - main
+ * - test.old
+ * - test.new
+ *
+ * The tests in `test.new` are compared for novelty against the tests in `test.old`.
+ *
+ * @property mavenProject The MavenProject instance to analyze.
+ */
+class NoveltyAnalyzer(private val mavenProject: MavenProject) {
+    companion object {
+        fun <T> getNumNovelSets(oldSets: Set<Set<T>>, newSets: Set<Set<T>>) = newSets.count { newSet ->
+            oldSets.none { oldSet -> oldSet.containsAll(newSet) }
+        }
+    }
+
+    private val logger = KotlinLogging.logger {}
```

Comment:
You can also get a logger by having your companion object extend `KLogging`. Just writing `companion object : KLogging() {` will make `logger` available as a (static) field.